### PR TITLE
Enable or disable features according to port configuration

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #329

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
The idea is great, but has some sense to define the port and disable the feature?

If not, to me, the correct way is to remove completely the "feature" in the firmware, and make it enable it simply when defined the port.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/tabs/ports.js
Comment (chmelevskij) [inline src/js/tabs/ports.js:429]: Something like that woudl work as well
```suggestion
            let functions = serialPortConfig.functions
            enableRxSerial = functions.includes('RX_SERIAL');
            enableTelemetry = functions.some(e => e.startsWith("TELEMETRY"));
            enableBlackbox = functions.includes('BLACKBOX');
            enableEsc = functions.includes('ESC_SENSOR');
            enableGps = functions.includes('GPS');
```

Comment (haslinghuis) [reply src/js/tabs/ports.js]: Thanks. Much easier to read.
Also @McGiverGim has a good point here.

Comment (haslinghuis) [reply src/js/tabs/ports.js]: @McGiverGim yes would make sense. Have to look at it

Comment (haslinghuis) [reply src/js/tabs/ports.js]: Also removed the promises.

**Review by chmelevskij — APPROVED** [review]

**Comment by haslinghuis** [comment]
Only some refactoring to reduce code size

**Review by limonspb — APPROVED** [review]

**Review by chmelevskij — APPROVED** [review]

**Review by McGiverGim — COMMENTED** [review]
As requested, I can approve it if you want, but as I said, I think this is not the correct approach. 

This is a workaround more that we need to maintain in the Configurator.

I think that if we want that something is enabled or disabled automatically, is better to remove the feature in the firmware and simply make it enabled by default. Or if we want to maintain the feature, enable it in the firmware not in the Configurator when it receives the MSP command.

The problem with the Configurator is that we need to maintain backward compatibility, if the things change between versions, we need to maintain old and new code. If we do this in the firmware part and it changes in a future, the old code is simply removed and new code replaces it.

**Comment by limonspb** [comment]
> As requested, I can approve it if you want, but as I said, I think this is not the correct approach.
> 
> This is a workaround more that we need to maintain in the Configurator.
> 
> I think that if we want that something is enabled or disabled automatically, is better to remove the feature in the firmware and simply make it enabled by default. Or if we want to maintain the feature, enable it in the firmware not in the Configurator when it receives the MSP command.
> 
> The problem with the Configurator is that we need to maintain backward compatibility, if the things change between versions, we need to maintain old and new code. If we do this in the firmware part and it changes in a future, the old code is simply removed and new code replaces it.

i would agree

**Comment by mituritsyn** [comment]
Could be VTX tab hided with the same logic in this PR? Discussed it with @limonspb a week ago and was going to open own PR

**Review by blckmn — APPROVED** [review]

**Pull request merged**
